### DISCUSSION TITLE
Docker image

### DISCUSSION THREAD

**Discussion by patpatpat123** [qltysh/qlty#1151] [Feature Requests]
Hello there,

Thanks for the release of the new tool qlty.

I went to search it on DockerHub and it seems it is still missing.

Could you please help publish a DockerHub version of qlty?

Thank you

**Comment by brynary** [comment]
@patpatpat123 thank you your message! I moved this over to GH Discussions where we track feature requests.

This is something that we will definitely consider. Overall, we are considering adding more distribution formats, including Homebrew.

Can you tell me a little bit more about the benefits of a Docker image for your use case?

**Comment by patpatpat123** [comment]
Hello @brynary ,

Thanks for moving this to discussion.

The Docker image use case is straightforward.

In our company, engineers can choose their laptop and OS, but we all have Docker.

So instead of having each and every engineer install its own version, its own configuration, we would like to have a common docker image with a docker compose type configuration.

From there, every engineer will have this as a baseline.

Is my description ok?

**Reply by brynary** [reply]
Yes, that is very helpful... So basically your team is using Docker and a Docker Compose file as a way to consistently use dev tooling across platforms. That makes sense.

I don't see any reason we shouldn't offer Docker images as a distribution format. We'll consider this planned, but we will probably add Homebrew packages first.

**Reply by patpatpat123** [reply]
You are correct.

I will just tag myself here and come back to check from time to time, hoping to see the docker version.

Thanks!

**Comment by Jellyfrog** [comment]
For CI/CD pipelines this is also useful, saves a lot of time if the image contains all dependencies/plugins already so they don't have to be downloaded on the fly.

**Comment by brynary** [comment]
We now have a Docker image available on GitHub Container Registry!

https://github.com/qltysh/qlty/pkgs/container/qlty

Usage:

```sh
# Install
docker pull ghcr.io/qltysh/qlty:main # Add --platform linux/x86_64 if needed

# Run, mounting current directory so source code is available
docker run --rm -it -v "$(pwd):/app" ghcr.io/qltysh/qlty:main help
```

This is brand new and should be considered experimental, so please let us know if you see any issues. Our official recommended installation method remains the executable binary installations.

If you would like to create an image with linters pre-installed, you should be able to do that with a Dockerfile that builds on this image as `FROM ghcr.io/qltysh/qlty:main`

**Reply by Jellyfrog** [reply]
> We now have a Docker image available on GitHub Container Registry!
> 
> https://github.com/qltysh/qlty/pkgs/container/qlty
> 
> Usage:
> 
> ```shell
> # Install
> docker pull ghcr.io/qltysh/qlty:main # Add --platform linux/x86_64 if needed
> 
> # Run, mounting current directory so source code is available
> docker run --rm -it -v "$(pwd):/app" ghcr.io/qltysh/qlty:main help
> ```
> 
> This is brand new and should be considered experimental, so please let us know if you see any issues. Our official recommended installation method remains the executable binary installations.
> 
> If you would like to create an image with linters pre-installed, you should be able to do that with a Dockerfile that builds on this image as `FROM ghcr.io/qltysh/qlty:main`

Nice! Still think it would be good to provide an official everything pre-installed image. Tag it with ”full” or similar.

**Reply by brynary** [reply]
Thanks!

Yes, I will create a new feature request to track the possibility of adding `*-full` variant images.

Aside from the that potential optimization, does this new Docker image work for you functionally?

(Also, I am curious, how much slowdown are you seeing from the linter installation phase? In our testing, it is typically fairly fast.)
